lms/djangoapps/instructor/lti_grader.py

```diff
@@ -0,0 +1,125 @@
+        self.secret = lti_secret
+        self.output = {'success': '', 'error': ''}
+
+    def _get_first_anon_id(self, grade_data):
+        """
+        Returns the first anonymized student id for in the given uploaded csv data file
+        :param grade_data: an InMemoryUploadedFile with lti grade data
+        :return: anon_id
+        """
+        reader = self._unicode_csv_reader(grade_data)
+        try:
+            reader.next()
+            first_row = reader.next()
+            anon_id = first_row[1]
+        except StopIteration:
+            self.output['error'] += '\nThe CSV file contains no useful data!'
+            anon_id = None
+        return anon_id
+
+    def update_grades(self, grade_data):
+        """
+        Updates grades for the LTI component with the grade data provided
+        :param grade_data: an InMemoryUploadedFile with lti grade data
+        :return: output, a dictionary with two keys: 'success' and 'error'
+        """
+        grade_data = grade_data.read().splitlines()
+
+        first_anon_id = self._get_first_anon_id(grade_data)
+        if first_anon_id is None:
+            return self.output
+
+        test_url = self.url_base + first_anon_id
+        try:
+            lti_connection.validate_lti_passport(self.key, self.secret, test_url)
+        except LTIConnectionError as error:
+            self.output['error'] += '\n' + unicode(error)
+            return self.output
+
+        for grade_row in self._generate_valid_grading_rows(grade_data):
+            try:
+                self.output['success'] += '\n' + lti_connection.post_grade(self.url_base, self.key, self.secret, grade_row)
+            except LTIConnectionError as error:
+                self.output['error'] += '\n' + unicode(error)
+        return self.output
+
+    def _generate_valid_grading_rows(self, data):
+        """
+        Yield valid grading rows from a file
+        """
+        reader = self._unicode_csv_reader(data)
+        row_num = 0
+        for row in reader:
+            row_num += 1
+
+            if row_num == 1:
+                continue
+
+            if len(row) < 5:
+                self.output['error'] += "\nBAD ROW: grading_csv row {row_number} ({row}) doesn't have enough info".format(
+                    row_number=row_num,
+                    row=row,
+                )
+                continue
+            try:
+                if len(row) == 5:
+                    yield tuple([
+                        int(row[0]),
+                        unicode(row[1]),
+                        unicode(row[2]),
+                        float(row[3]),
+                        float(row[4]),
+                    ])
+                else:
+                    yield tuple([
+                        int(row[0]),
+                        unicode(row[1]),
+                        unicode(row[2]),
+                        float(row[3]),
+                        float(row[4]),
+                        unicode(row[5]),
```

Comment:
Does leaving the comments field blank get read in by csv reader as empty string?